src/activity_actor.cpp

```diff
@@ -2351,6 +2351,50 @@ void reload_activity_actor::finish( player_activity &act, Character &who )
     } else {
         add_msg( m_neutral, _( "You reload the %1$s with %2$s." ), reloadable_name, ammo_name );
     }
+    item_location loc = reload_targets[0];
+    tripoint point = loc.position();
+    // Reload may have caused the item to increase in size more than the pocket can contain.
+    // We want to avoid this because items will be deleted on a save/load.
+    if( loc.where() == item_location::type::container &&
+        loc.held_by( who ) ) {
+        item_pocket pocket = reloadable.get_containing_pocket( who );
+        if( pocket.remaining_volume() < units::volume() ||
+            pocket.remaining_weight() < units::mass() ) {
+            if( !who.is_armed() ) {
+                add_msg( m_neutral, _( "The %s no longer fits in your inventory so you wield it instead." ),
+                         reloadable_name );
+                who.wield( reloadable );
+            } else {
+                item replace = reloadable;
+                who.remove_item( reloadable );
+                add_msg( m_neutral, _( "The %s no longer fits in your inventory so you drop it instead." ),
+                         reloadable_name );
+                get_map().add_item_or_charges( point, replace );
+            }
+        }
+        // Check if in a vehicle.
+    } else if( loc.where() == item_location::type::vehicle ) {
+        const cata::optional<vpart_position> vp = get_map().veh_at( point );
```

Comment:
Same here, you can get the vehicle and part index from the `item_location` directly.

Reply:
Ok will take a look.